homeassistant/components/accuweather/weather.py

```diff
@@ -86,52 +100,60 @@ def condition(self):
             return None
 
     @property
-    def temperature(self):
+    def temperature(self) -> StateType:
         """Return the temperature."""
-        return self.coordinator.data["Temperature"][self._unit_system]["Value"]
+        return cast(
+            float, self.coordinator.data["Temperature"][self._unit_system]["Value"]
+        )
 
     @property
-    def temperature_unit(self):
+    def temperature_unit(self) -> str:
         """Return the unit of measurement."""
         return TEMP_CELSIUS if self.coordinator.is_metric else TEMP_FAHRENHEIT
 
     @property
-    def pressure(self):
+    def pressure(self) -> StateType:
         """Return the pressure."""
-        return self.coordinator.data["Pressure"][self._unit_system]["Value"]
+        return cast(
+            float, self.coordinator.data["Pressure"][self._unit_system]["Value"]
+        )
 
     @property
-    def humidity(self):
+    def humidity(self) -> StateType:
         """Return the humidity."""
-        return self.coordinator.data["RelativeHumidity"]
+        return cast(int, self.coordinator.data["RelativeHumidity"])
 
     @property
-    def wind_speed(self):
+    def wind_speed(self) -> StateType:
         """Return the wind speed."""
-        return self.coordinator.data["Wind"]["Speed"][self._unit_system]["Value"]
+        return cast(
+            float, self.coordinator.data["Wind"]["Speed"][self._unit_system]["Value"]
+        )
 
     @property
-    def wind_bearing(self):
+    def wind_bearing(self) -> StateType:
         """Return the wind bearing."""
-        return self.coordinator.data["Wind"]["Direction"]["Degrees"]
+        return cast(int, self.coordinator.data["Wind"]["Direction"]["Degrees"])
 
     @property
-    def visibility(self):
+    def visibility(self) -> StateType:
```

Comment:
It's allowed to use more narrow type here.
```suggestion
    def visibility(self) -> float:
```